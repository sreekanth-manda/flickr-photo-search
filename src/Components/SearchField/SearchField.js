import React from 'react';
import './SearchField.scss';

const SearchField = ({ searchText, onSearchInputChange }) => (
  <input
    type="text"
    className="search-input"
    value={searchText}
    onChange={onSearchInputChange}
    placeholder="Search images here"
  />
);

export default SearchField;