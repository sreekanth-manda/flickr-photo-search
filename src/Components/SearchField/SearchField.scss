.search-input {
  margin: 10px;
  width: 80%;

  @media screen and (min-width: 600px) {
      width: 500px;
  }
}