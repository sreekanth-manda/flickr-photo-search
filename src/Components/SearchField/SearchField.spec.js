import React from 'react';
import { shallow } from 'enzyme';
import SearchField from './SearchField';

const onSearchInputChange = jest.fn();

describe('SearchField', () => {
  const compoentProps = {
    searchText: 'blah',
    onSearchInputChange
  }

  test('should render SerachField component', () => {
    const wrapper = shallow(<SearchField {...compoentProps} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should call onSearchInputChange when some text is typed into the search box', () => {
    const wrapper = shallow(<SearchField {...compoentProps} />);
    wrapper.find('input').simulate('change', { target: { value: 'Hello' } })
    expect(onSearchInputChange.mock.calls.length).toBe(1);
  })

  test('should have a place holder text on search box', () => {
    const wrapper = shallow(<SearchField {...compoentProps} />);
    expect(wrapper.find('input').prop('placeholder')).toBe('Search images here');
  })
});