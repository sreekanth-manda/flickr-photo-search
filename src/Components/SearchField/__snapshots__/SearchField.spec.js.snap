// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField should render SerachField component 1`] = `"<input type=\\"text\\" class=\\"search-input\\" value=\\"blah\\" placeholder=\\"Search images here\\"/>"`;
